### PR TITLE
query and lock timeout on various databases

### PR DESCRIPTION
**Pull request by njr-11**
Implement code around how query and lock timeout are established via the JPA persistence service.  Update tests for coverage with the set of databases that we typically test with.

During this effort, it was discovered that the mechanism for detecting the Oracle JDBC isn't working, and needed to be replaced.

Observed default behavior for various databases:
Oracle and PostgreSQL - timeout is unnecessary because they return from the query based on the committed state rather than being blocked by pending updates.
SQL Server - can get lock timeout immediately when EclipseLink generates SQL like: SELECT ID FROM WLPTASK WITH (UPDLOCK, NOWAIT) WHERE ((ID = ?) AND (VERSION = ?))
DB2 - blocked until query timeout occurs
Derby - does not appear to support query timeout or a more granular lock timeout.  The global 1 minute lock timeout applies, which causes a long wait.

### CODE REVIEW

File: dev/com.ibm.ws.concurrent.persistent_fat/test-applications/schedtest/src/web/SchedulerFATServlet.java [hunk 1]
```diff
@@ -80,6 +84,11 @@
      */
     private static final String SUCCESS_MESSAGE = "COMPLETED SUCCESSFULLY";
 
+    /**
+     * Indicates if the database for the persistent executor is Derby.
+     */
+    private boolean isDerby;
```
Comment (KyleAure) [inline dev/com.ibm.ws.concurrent.persistent_fat/test-applications/schedtest/src/web/SchedulerFATServlet.java]: You could instead use: 
```
private final boolean isDerby = System.getProperty("fat.bucket.db.type", "Derby").compareToIgnoreCase("Derby") == 0;
```
And avoid making a connection in the `init` method

Comment (njr-11) [reply dev/com.ibm.ws.concurrent.persistent_fat/test-applications/schedtest/src/web/SchedulerFATServlet.java]: That's a nice improvement. I'll update it.

**Comment by njr-11** [comment]
#run-libby-bot
#build

**Comment by njr-11** [comment]
#build

**Comment by njr-11** [comment]
#build

**Comment by njr-11** [comment]
#build

**Comment by njr-11** [comment]
Build has 1 test failure, which is a pre-existing issue in ui tests

**Pull request merged**